### PR TITLE
Added post about Enumerable#grep vs Enumerable#select.

### PR DESCRIPTION
**Pull request by mauro-oto**
Hey @etagwerker, 

This PR adds a post about `Enumerable#grep` vs `Enumerable#select`. 

Please check it out and let me know what you think, thanks!

### CODE REVIEW

File: _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+over `#select`.
+
+See the benchmark below for example, which uses the class matcher for `#grep`:
+
+```ruby
+2.1.2 :041 > a = (1..100000).map { rand(100000) }
+2.1.2 :042 > b = (1..100000).map { rand(100000).to_s }
+2.1.2 :043 > c = a + b
+2.1.2 :044 > Benchmark.bm(10) do |b|
+2.1.2 :045 >   b.report("select")    { c.select { |x| x.to_i == x } }
```
Comment (etagwerker) [inline _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown]: @mauro-oto I think that the benchmark should compare "equal" comparisons, to make it more objective.

Comment (mauro-oto) [inline _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown]: Yeah, I was thinking the same. Maybe compare x.class == Integer? I'm not sure how to replicate what `#grep` is doing there with the class comparison. Or I could just leave the other benchmark, and comment how `#grep` can use the class matcher.

Comment (etagwerker) [inline _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown]: Maybe you should  use `#is_a?` http://ruby-doc.org/core-2.2.1/Object.html#method-i-is_a-3F

Comment (mauro-oto) [inline _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown]: Awesome, that'll do it, thanks!

File: _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+                 user     system      total        real
+select       0.070000   0.000000   0.070000 (  0.064458)
+grep         0.020000   0.000000   0.020000 (  0.023958)
+
+2.1.2 :048 > c.select { |x| x.to_i == x } == c.grep(Integer)
+ => true
+```
+
+You can also see its power when using Regexp patterns, which is not only faster
+but also less verbose, which is enough of a win to favor it over `#select`:
```
Comment (etagwerker) [inline _posts/2015-01-19-enumerator-grep-vs-enumerator-select.markdown]: :+1:

**Comment by etagwerker** [comment]
Nice article!

**Comment by mauro-oto** [comment]
@etagwerker Thanks! I changed the benchmark to use `is_a?`, which is faster but `grep` still outperforms `select`.

**Pull request merged**